### PR TITLE
Add section for basic and advanced directives usage

### LINKED ISSUE #521: Section on Directives
We cover components, but not `@Directive`

* [ ] - Finalize section outline
* [ ] - Attribute Directives
* [ ] - Host Listener
* [ ] - Host Binding
* [ ] - Structural Directives

### PR DESCRIPTION
**Pull request by jamesbs**
Fixes #521 
- Includes basic usage section describing attribute and structural directives and Angular 2's built in versions of these.
- Updated SUMMARY.md to include the basic directives section after components since the built in directives are a large part of basic Angular 2 template syntax
- Removed the old structural directives section since the new section covers that content
- Added Advanced Angular section describing how to create custom attribute and structural directives
- Includes mention to HostListener, TemplateRef, ViewContainerRef and embedded views
- Host Binding not mentioned. Since HostBinding can also apply to components, I feel this is better mentioned in an advanced topic of its own.

### CODE REVIEW

File: handout/advanced-angular/directives/creating_a_structural_directive.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Creating a Structural Directive
```
Comment (e-schultz) [inline handout/advanced-angular/directives/creating_a_structural_directive.md]: Wondering if this should just be under the directives section?

File: handout/advanced-angular/directives/creating_an_attribute_directive.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
+# Creating an Attribute Directive
```
Comment (e-schultz) [inline handout/advanced-angular/directives/creating_an_attribute_directive.md]: I like how you walk through this, 👍

**Comment by e-schultz** [comment]
All around I like the content of the sections so far, just wondering if we need to have it be "Advanced Angular", or simply put the creating your own directives section underneath the directives.

**Comment by jamesbs** [comment]
I put it in a separate section because I felt it would be a better flow to go over the core parts of an Angular 2 application: _components, directives, DI, change detection_ before going over topics that put those parts together. It seemed too early to put in a section that talks about injecting ViewContainerRef without talking about what Angular 2 DI is and how it works. I also wanted to cover the basic directives like ngIf right after components since a lot of component work interacts with the template and basic template usage will often involve the built in Angular directives. Thoughts?

**Comment by e-schultz** [comment]
@jamesbs can you fix the merge conflicts on this?

**Pull request closed without merging**